application/helpers/update/updates/Update_607.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace LimeSurvey\Helpers\Update;
+
+use Exception;
+
+/**
+ * Fix organizer link : icon and survey activated
+ * @package LimeSurvey\Helpers\Update
+ */
+class Update_607 extends DatabaseUpdateBase
+{
+    /**
+     * @inheritDoc
+     * @throws \CException
+     */
+    public function up()
+    {
+        $surveyThemes = $this->db->createCommand()
+            ->select('*')
+            ->from('{{templates}}')
+            ->queryAll();
+        if (!empty($surveyThemes)) {
+            foreach ($surveyThemes as $surveyTheme) {
+                $surveyThemeName = $surveyTheme['name'];
+                setTransactionBookmark();
+                try {
+                    if (!$this->isStandardTemplate($surveyThemeName)) {
+                        $rowsDeleted = $this->db->createCommand()->delete(
+                            '{{templates}}',
+                            'name=:name',
+                            [':name' => $surveyThemeName]
+                        );
+                        if ($rowsDeleted >= 1) {
+                            $this->db->createCommand()->delete(
+                                '{{template_configuration}}',
+                                'template_name=:templateName',
+                                [':templateName' => $surveyThemeName]
+                            );
+                        }
+                    }
+                } catch (Exception $e) {
+                    rollBackToTransactionBookmark();
+                }
+            }
+        }
+    }
+
+    /**
+     * Check if a survey theme is a core theme
+     * @param $surveyThemeName
+     * @return bool
+     */
+    private function isStandardTemplate($surveyThemeName): bool
```

Comment:
Couldn't this be just a hard-coded list?

Reply:
Well i could just check against specific themenames that are standardthemes and  in the table.